python/config_settings/config_settings.bzl

```diff
@@ -29,16 +29,19 @@ def construct_config_settings(name, python_versions):
 
     # Maps e.g. "3.8" -> ["3.8.1", "3.8.2", etc]
     minor_to_micro_versions = {}
-    allowed_flag_values = []
+
+    # Add the legacy value that is used in non-version aware toolchains
+    allowed_flag_values = ["PY3"]
```

Comment:
@rickeylev can you think of any unintended consequences here? If it looks good to you feel free to press merge.

Reply:
Is the idea to support a more drop-in replacement of `py_binary(python_version="PY3")` with a version-aware `py_binary` symbol?

I think that's a good idea, but it makes more sense to drop the value in the transition rather than accept the value as flag state. i.e. in the transition:

```
def transition_ipml(_, attr):
  pv = attr.python_version
  if pv == in ("PY3", "PY3ONLY", ...):
    pv = ""  # default, or some such
  return {"...:python_version": pv}
```

Then nothing ever sees "PY3" as value and we don't have to worry about people trying to match PY3 with config_setting.flag_values

That's the main unintended consequence I see here -- people will try to match on that flag value as some indicator of "wasn't set" or "unspecified version" or something